### DISCUSSION TITLE
JobRunr Recurring Job Starts Early

### DISCUSSION THREAD

**Discussion by pjacobs** [jobrunr/jobrunr#809] [General]
I am finding that recurring jobs sometimes start up to 8 seconds earlier than scheduled.  Is this by design?  Here is an example of what I have discovered.  The attached screenshot shows a bunch of test jobs I have scheduled via cron.  There were scheduled via the sample code below

`BackgroundJob.scheduleRecurrently("test-job2", "0 18 2 * * *", this::test);`

`public void test() {
        logger.debug("Starting test job!");
    }`

So, I would expect `test()` to be executed at 2:18:00.  However, in the log I am seeing this

`2023-07-26T02:17:53.132+09:00 DEBUG 46117 --- [pool-2-thread-5] c.a.g.c.j.JobRunrConfig$$SpringCGLIB$$0  : Starting test job!`

seemingly implying that the job started nearly 7 seconds early.  Am I perhaps doing something wrong when scheduling the job?  I have also attached a screenshot showing the dashboard.
![Screenshot 2023-07-26 at 2 18 51](https://github.com/jobrunr/jobrunr/assets/333548/f57f97ff-bb38-4a67-8151-4066d8bc9b6e)
  Thanks in advance for taking the time to answer my question.  I realize that you put in your spare time with the jobrunr project.

**Comment by rdehuyss** [comment]
This is mentioned in the docs...

**Reply by rdehuyss** [reply]
See https://www.jobrunr.io/en/documentation/background-methods/recurring-jobs/ for more info.

**Reply by pjacobs** [reply]
Thank you for your response and apologies for missing it in the docs.
